packages/suite/src/hooks/wallet/useSendFormFields.ts

```diff
@@ -11,86 +17,147 @@ import { SendContextValues, UseSendFormState } from 'src/types/wallet/sendForm';
+            const outputError = errors.outputs ? errors.outputs[outputId] : undefined;
+            const error = outputError
+                ? outputError[target === 'fiat' ? 'amount' : 'fiat']
+                : undefined;
+            if (error || !value) {
+                if (targetValue.length > 0) {
+                    setValue(targetInputName, '');
+                    clearErrors(targetInputName);
                 }
 
                 return;
             }
-            const fiatRateKey = getFiatRateKey(
-                network.symbol,
-                currency.value as FiatCurrencyCode,
-                token as TokenAddress,
-            );
-            const fiatRate = currentRates?.[fiatRateKey];
-            // calculate Fiat value
-            if (!fiatRate?.rate) return;
-
-            const formattedAmount = shouldSendInSats // toFiatCurrency always works with BTC, not satoshis
-                ? formatNetworkAmount(amount, network.symbol)
-                : amount;
-
-            const fiatValue = toFiatCurrency(formattedAmount, fiatRate.rate, 2);
-            if (fiatValue) {
-                setValue(inputName, fiatValue, { shouldValidate: true });
+
+            const fiatRate = getCurrentFiatRate({
+                currencyCode: output.currency.value as FiatCurrencyCode,
+                tokenAddress: output.token as TokenAddress,
+            });
+            if (!fiatRate?.rate) {
+                return;
+            }
+            const formattedTargetValue = formatTargetValue(value, fiatRate.rate);
+            if (formattedTargetValue) {
+                setValue(targetInputName, formattedTargetValue, { shouldValidate: true });
             }
         },
-        [getValues, setValue, currentRates, shouldSendInSats, network.symbol, errors],
+        [clearErrors, getCurrentFiatRate, getValues, setValue, errors],
+    );
+
+    const calculateFiatFromAmount = useCallback(
+        (outputId: number, amount: string) => {
+            const calculateFormattedFiatValue = (amount: string, fiatRate: number) => {
+                const formattedAmount = shouldSendInSats // toFiatCurrency always works with BTC, not satoshis
+                    ? formatNetworkAmount(amount, network.symbol)
+                    : amount;
+
+                return toFiatCurrency(formattedAmount, fiatRate, 2);
+            };
+
+            return calculateFiatFromAmountOrViceVersa({
+                formatTargetValue: calculateFormattedFiatValue,
+                outputId,
+                target: 'fiat',
+                value: amount,
+            });
+        },
+        [calculateFiatFromAmountOrViceVersa, shouldSendInSats, network.symbol],
+    );
+
+    const calculateAmountFromFiat = useCallback(
+        (outputId: number, fiat: string) => {
+            const calculateFormattedAmountValue = (fiat: string, fiatRate: number) => {
+                const amount = fromFiatCurrency(fiat, network.decimals, fiatRate);
```

Comment:
the original implementation used `token.decimals || network.decimals` ..?

Reply:
Nice catch! c49af7248472705a6f9e90cdabaecce8d08fd2a4